src/duckdb.mjs

```diff
@@ -1,361 +1,228 @@
-import {
-  arrow as arr,
-  duckdb as duck
-} from "./dependencies.mjs";
-
-export default async function duckdb(require) {
-
-  const arrow = await require(arr.resolve()); // TODO is this right...?
-  const bundles = await duck.getJsDelivrBundles();
-  const bundle = await duck.selectBundle(bundles);
-  async function makeDB() {
-    const logger = new duck.ConsoleLogger();
-    const worker = await duck.createWorker(bundle.mainWorker);
-    const db = new duck.AsyncDuckDB(logger, worker);
-    await db.instantiate(bundle.mainModule);
-    return db;
+import {arrow9 as arrow, duckdb} from "./dependencies.mjs";
+import {FileAttachment} from "./fileAttachment.mjs";
+
+// Adapted from https://observablehq.com/@cmudig/duckdb-client
+// Copyright 2021 CMU Data Interaction Group
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+//    this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// TODO Allow this to be overridden using the Library’s resolver.
+const cdn = "https://cdn.observableusercontent.com/npm/";
+
+export class DuckDBClient {
+  constructor(db) {
+    Object.defineProperties(this, {
+      _db: {value: db}
+    });
   }
 
-  // Adapted from: https://observablehq.com/@cmudig/duckdb-client
-  // Follows the DatabaseClient specification: https://observablehq.com/@observablehq/database-client-specification
-  class DuckDBClient {
-    constructor(_db) {
-      this._db = _db;
-      this._counter = 0;
+  async queryStream(query, params) {
+    const connection = await this._db.connect();
```

Comment:
What type of object is `this._db` that has the `.connect` method?

Reply:
A database client? Not seeing a `.connect` method in the [specification](https://observablehq.com/@observablehq/database-client-specification)

Reply:
No, it is an AsyncDuckDB instance created here:

https://github.com/observablehq/stdlib/blob/a0fef0c392cbfa67ada6a65d91ddad46f11b3280/src/duckdb.mjs#L120

The connect method is documented here:

https://shell.duckdb.org/docs/classes/index.AsyncDuckDB.html#connect